File: api/routes/degree.ts
import express from 'express';

import { degreeSearch, degreeById, degreeInsert } from '../controllers/degree';

const router = express.Router();

router.get('/search?', degreeSearch);
router.get('/:id', degreeById);
router.post('/insert', degreeInsert);

export default router;

File: api/routes/degree.go
package routes

import (
	"github.com/gin-gonic/gin"

	"github.com/UTDNebula/nebula-api/api/controllers"
)

func DegreeRoute(router *gin.Engine) {
	// All routes related to degrees come here
	degreeGroup := router.Group("/degree")

	degreeGroup.OPTIONS("", controllers.Preflight)
	degreeGroup.GET("", controllers.DegreeSearch())
	degreeGroup.GET(":id", controllers.DegreeById())
}
